pvlib/clearsky.py

```diff
@@ -597,6 +599,264 @@ def _calc_d(aod700, p):
+def _calc_slope_nstd(data):
+    return np.diff(data).std(ddof=1) / data.mean()
+
+
+def _shift_from_align(align, window):
+    # number of places to shift to line up pandas.Series.rolling with
+    # align value.
+    # pandas.Series.rolling uses kwarg center with values of True or False
+    # default is False which means right aligned labels by default
+    # center=True aligns to center
+    # Here, calculate shift to align='left' (legacy pvlib behavior)
+    # code here for future capability if align='left', 'right' are of interest
+    # commented out to avoid decreasing test coverage
+    # if align == 'left':
+    #     shift = 1 - window
+    # else:
+    #     shift = 0
+    shift = 0
+    return shift
+
+
+def _count_in_window(ts, window):
+    # return int type instead of default int64, to avoid error in pandas
+    return int(ts.resample(window).agg('sum')[0])
+
+
+def _get_sample_intervals(times, window_length):
+    """ Calculates timeinterval and samples per window for Reno-style clear
+    sky detection functions
+    """
+    window = pd.Timedelta(minutes=window_length)
+    temp = pd.Series(data=1, index=times)
+
+    # determine if we can proceed
+    if times.inferred_freq and len(temp.unique()) == 1:
+        sample_interval = times[1] - times[0]
+        sample_interval = sample_interval.seconds / 60  # interval in minutes
+        samples_per_window = _count_in_window(temp, window)
+        return sample_interval, samples_per_window
+    else:
+        raise NotImplementedError('algorithm does not yet support unequal '
+                                  'times. consider resampling your data.')
+
+
+def _calc_line_length(data, sample_interval):
+    """ Calculates line length for Reno-style clear sky detection functions.
+    """
+    return np.sum(np.sqrt(np.diff(data)**2 + sample_interval**2))
+
+
+def _maxdiff(s):
+    return np.abs(np.diff(s)).max()
+
+
+def _calc_c5(meas, clear, window, align, limit):
+    center = align == 'center'
+    irrad_diff = meas - clear
+    slope_diff = irrad_diff.rolling(window, center=center).apply(_maxdiff)
+    return slope_diff < limit
+
+
+def _clear_sample_index(clear_windows, samples_per_window, align, H):
+    """
+    Returns indices of clear samples in clear windows
+    """
+    # first window is in first column of H. clear_windows is aligned by 'align'
+    # shift clear_windows.index so that first window is in left position, lined
+    # up with column of H containing the indices of the first window
+    # commented if/else block for future align='left', 'right' capability
+    # if align == 'right':
+    #     shift = 1 - samples_per_window
+    # elif align == 'center':
+    #     shift = - (samples_per_window // 2)
+    # else:
+    #     shift = 0
+    shift = -(samples_per_window // 2)
+    idx = clear_windows.shift(shift)
+    # drop rows at the end corresponding to windows past the end of data
+    idx.drop(clear_windows.tail(samples_per_window - 1).index, inplace=True)
```

Comment:
Is the argument here equivalent to `clear_windows.index[samples_per_window-1:]`?

Reply:
Equivalent to `clear_windows.index[1 - samples_per_window:]` is there a reason to prefer this over `tail`?

Reply:
A bit more efficient (py3.7, pandas 0.25.1) but I won't raise a fuss about it.
```python
s = pd.Series(range(8760), index=pd.date_range('2019-01-01', freq='h', periods=8760))

%timeit _ = s.tail(10).index
241 µs ± 10.3 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

%timeit _ = s.index[-10:]
134 µs ± 4.23 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```